pkg/store/container/metadata.go

```diff
@@ -34,10 +34,12 @@ const metadataVersion = "v1" // nolint
 // nolint
 type versionedMetadata struct {
 	// Version indicates the version of the versioned container metadata.
-	Version string
-	Metadata
+	Version  string
+	Metadata metadataInternal
```

Comment:
Why do we need this?

Reply:
There is a recursive call  if we use Metadata in versionedMetadata derectly.

Reply:
Make sense.